csrc/includes/context.h

```diff
@@ -64,17 +64,18 @@ class Context {
         return _ctx;
     }
 
-    void GenWorkSpace(size_t size)
+    void GenWorkSpace(void* workspace)  // (size_t size)
     {
-        if (!_workspace) {
+        if (!workspace) { throw std::runtime_error("Workspace is null."); }
+        _workspace = workspace;
+        /*if (!_workspace) {
```

Comment:
remove those commented code if they are useless.

Reply:
I have just commented them to verify that this new way is working properly, after some tests I will remove the comments